internal/fs/buffer/inmemory_write_buffer.go

```diff
@@ -15,33 +15,165 @@
 package buffer
 
 import (
-	"bytes"
+	"fmt"
+
+	"github.com/googlecloudplatform/gcsfuse/internal/logger"
 )
 
+type offset struct {
+	// [ start, end) is the range of file offset for which the
+	// data is currently stored in the buffer.
+	start, end int64
+}
+
+func (o *offset) advanceBy(value int64) {
+	o.start += value
+	o.end += value
+}
+
+type inMemoryBuffer struct {
+	buffer []byte
+	offset offset
+}
+
+// InMemoryWriteBuffer implements WriteBuffer and is stored in memory.
+// Refer to buffer.WriteBuffer for sample code usage.
 type InMemoryWriteBuffer struct {
```

Comment:
This struct need not be public. This struct by itself isn't needed anywhere, only the interface WriteBuffer is. 

This is not very different from https://github.com/GoogleCloudPlatform/gcsfuse/blob/3679762e4ed291a26ae73c471abfc22d57799227/internal/storage/chunk_uploader.go#L37 .
If we just change `CreateInMemoryWriteBuffer` to return WriteBuffer (similar to https://github.com/GoogleCloudPlatform/gcsfuse/blob/3679762e4ed291a26ae73c471abfc22d57799227/internal/storage/chunk_uploader.go#L55), instead of `*InMemoryWriteBuffer`, it'll all fall in place.